src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDConfigurationTests.cs

```diff
@@ -948,6 +951,9 @@ public void CSharp_VerifyIDECodeStyleOptionsAreConfigurable()
 # IDE0083, PreferNotPattern
 csharp_style_prefer_not_pattern = true
 
+# IDE0085, PreferNamespaceMatchFolderStructure
+cshar_namespace_match_folder_structure_preference = true
```

Comment:
1. `cshar`
2. this doesn't seem cshar specific.  would expect this to be dotnet_xxx
3. this does't seem like a true/false preference.  for example, i could see it more like "the analyze should try to make the files match teh namespace structure, not vice versa"
4. do we generally name these with _preference?

Reply:
Changed to `csharp_style_namespace_match_folder_structure` 

This is currently only a csharp feature. Implementation of code fix prevents vb right now. If it should still be dotnet_* even if we don't (and may never? depending on priority etc) have a solution for vb I'm happy to move.

Reply:
> Implementation of code fix prevents vb right now.

Can you clarify why that is?

> If it should still be dotnet_* even if we don't

1. Yes, i think this should be dotnet_.  This is because there's nothing truly C# specific here.  i.e. this isn't about "show var or not" when VB doesn't even have that concept :)
2. `and may never?` I truly hope that won't be the case.  That definitely violates some team goals, and stated views on VB's place alongside C#.  We've been pretty adamant that our tooling efforts care about both (unless there is a compelling case to not do it).  
3. I would think there woudl be very little truly c# specific behavior here.  VB has some interesting lang aspects (like a default namespace).  However, C# may be getting that as well.  So i would want us to be able to just support both these languages as close to equally as we can.

Reply:
> Can you clarify why that is? 

This depends heavily on the [sync namespace feature](https://docs.microsoft.com/en-us/visualstudio/ide/reference/sync-namespace-and-folder-name?view=vs-2019) which is only c# for now. It's an example of feature code that was written and expanded to multiple scenarios, but hasn't been implemented for VB. 

> and may never...

This wasn't a statement of goals on the team, but a bit of pessimism on my part. It's in the backlog, but afaik there hasn't been any driving factor to change the priority. Once we have a VB implementation of "sync namespace" we can move forward with a VB implementation here. I think https://github.com/dotnet/roslyn/issues/31253 tracks remaining work? @genlu  might know more. 

I'll rename to "dotnet_" and add language agnostic abstraction for now, even if the initial implementation is only c#.

Reply:
> This depends heavily on the sync namespace feature which is only c# for now.

I was not aware of that.  Is there a reason we didn't do this for VB?  From my perspective, we need a *really* good reason to not make our feature available for both languages.

Reply:
https://github.com/dotnet/roslyn/pull/30910 sources https://github.com/dotnet/vblang/issues/351 as the reason. I haven't read the discussion yet, but sharing what I found.